src/flatpakManifestManager.ts

```diff
@@ -0,0 +1,191 @@
+import { FlatpakManifest } from './flatpakManifest';
+import { FlatpakManifestFinder } from './flatpakManifestFinder';
+import { exists } from './utils';
+import { promises as fs } from 'fs'
+import * as vscode from 'vscode'
+import { WorkspaceState } from './workspaceState';
+import { EXT_ID } from './extension';
+
+export interface ManifestQuickPickItem {
+    label: string,
+    detail: string,
+    manifest: FlatpakManifest
+}
+
+export class FlatpakManifestManager extends vscode.Disposable {
+    private readonly statusItem: vscode.StatusBarItem
+    private readonly workspaceState: WorkspaceState
+    private readonly finder = new FlatpakManifestFinder()
+    private readonly manifestWatcher: vscode.FileSystemWatcher
+    private manifests?: Set<FlatpakManifest>
+    private activeManifest: FlatpakManifest | null
+
+    private readonly _onDidActiveManifestChanged = new vscode.EventEmitter<[FlatpakManifest | null, boolean]>()
+    readonly onDidActiveManifestChanged = this._onDidActiveManifestChanged.event
+
+    constructor(workspaceState: WorkspaceState) {
+        super(() => this.dispose())
+
+        this.workspaceState = workspaceState
+        this.activeManifest = null
+
+        this.manifestWatcher = vscode.workspace.createFileSystemWatcher(this.finder.getManifestGlobPattern())
+        this.manifestWatcher.onDidCreate(async (newUri) => {
+            console.log(`Possible manifest created at ${newUri.fsPath}`)
+            const newManifest = await this.finder.parseManifest(newUri)
+            if (newManifest !== null) {
+                this.manifests?.add(newManifest)
+            }
+        })
+        this.manifestWatcher.onDidChange(async (uri) => {
+            console.log(`Possible manifest modified at ${uri.fsPath}`)
+            const activeManifest = this.getActiveManifest()
+            if (uri.fsPath !== activeManifest?.uri.fsPath) {
```

Comment:
Do we want to also update the non active manifests too?

Reply:
Currently, it may produce unexpected behavior when the user modified an unactive manifest then switch to it.